### PR TITLE
Forward compatibility for Commons IO 2.13.0 in `hudson.Util#loadFile`

### PR DESCRIPTION
**Pull request by basil**
`hudson.Util#loadFile` has some peculiar behavior that is documented in code comments but not covered by tests. Unfortunately, upgrading Commons IO to 2.13.0 breaks this behavior, but does so without any test failures. Fortunately I discovered this through code inspection and wrote a test that passes against the current version of Commons IO and fails against Commons IO 2.13.0.

With that having been done, I improved the implementation in `src/main` so that it did not depend (as much) on the implementation details of either Commons IO or the Java Platform but instead achieved the desired behavior by explicitly specifying character set decoding options. This will make this method less fragile. The new implementation passes the test I wrote with both the current version of Commons IO and Commons IO 2.13.0. As such I am checking in this change as a prerequisite to upgrading Commons IO 2.13.0. This PR itself does not upgrade Commons IO but rather prepares us for that upgrade.

### Testing done

- Current implementation in `src/main`, new test in `src/test`, current version of Commons IO: Pass (as expected)
- Current implementation in `src/main`, new test in `src/test`, new version of Commons IO: Fail (as expected)
- New implementation in `src/main`, new test in `src/test`, current version of Commons IO: Pass (as expected)
- New implementation in `src/main`, new test in `src/test`, new version of Commons IO: Pass (as expected)

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8142"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
